apps/emqx_resource/src/emqx_resource.erl

```diff
@@ -255,11 +259,23 @@ query(ResId, Request) ->
 -spec query(resource_id(), Request :: term(), emqx_resource_worker:query_opts()) ->
     Result :: term().
 query(ResId, Request, Opts) ->
+    {BridgeType, _BridgeName} = emqx_bridge_resource:parse_bridge_id(ResId),
+    %% TODO: this is a call layer violation
+    %% bridge is a hiher level abstraction of resource
+    %% resource should never call back up to bridge layer
+    %% Solution: pass down #{id => Id, name => Name, resource_type => ResourceType}
+    %%           instead of just ResId
+    ResourceType = emqx_bridge_resource:bridge_to_resource_type(BridgeType),
```

Comment:
The `emqx_resource_manager:ets_lookup(ResId)` will return the callback module, so:
```
case emqx_resource_manager:ets_lookup(Id) of
   {ok, _Group, #{mod := Mod,  ..} = Data} ->
      IsBufferSupported = is_buffer_supported(Mod)
end.
```